### ISSUE TITLE
Improve the base repository's Update method implementation.

### ISSUE THREAD

**Issue by fiseni** [ardalis/Specification#203]
The update method is implemented as follows

```c#
public virtual async Task UpdateAsync(T entity, CancellationToken cancellationToken = default)
{
	dbContext.Entry(entity).State = EntityState.Modified;

	await SaveChangesAsync(cancellationToken);
}
```

Here we set the state manually and that's not nice. It doesn't work if the entity has owned types, and generally, it's not a correct approach. Instead, we should utilize the `DbContext.Update` method. It ensures that all edge cases are covered, and marks the entity and all related types as modified.

It's important to emphasize that if we set the state to modified (manually or through the DbContext.Update() method), EF will consider the whole entity as modified, and regardless if you changed only one property, EF will produce SQL Update statements containing all columns. Practically we're not making use of the EF's change tracker. Saving changes is more than enough, EF will figure out what is changed and produce a SQL Update statement accordingly. There is a caveat here, and it all depends on how you actually use EF. If you're fetching the entity, mutating a state, and saving, then it's all good. But, if you try to update an entity that is not tracked by EF, then obviously it won't work. If we want to cover these cases, then we should utilize the dbContext.Update() method.

To summarize, we should update the repository method as follows

```c#
public virtual async Task UpdateAsync(T entity, CancellationToken cancellationToken = default)
{
    dbContext.Set<T>().Update(entity);

    await SaveChangesAsync(cancellationToken);
}
```

Usage recommendation:
- If you're loading an entity, modifying it, and you want to update it - Call the repository's `SaveChangesAsync` method. Don't call the repository's `Update` method at all.
- If you manually create the entity to be updated - Call the repository's `Update` method. It will include the newly created entity to the tracker (as modified) and will save the changes.

**Comment by ardalis** [comment]
Note: This only works with EF Core, not EF 6
